evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -240,6 +248,47 @@ fn remove_file(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes,
     Ok(Bytes::new())
 }
 
+/// Find the ABI type of a JSON Value
+fn value_to_type(value: &Value) -> Option<ParamType> {
+    if value.is_boolean() {
+        Some(ParamType::Bool)
+    } else if value.is_string() {
+        // If string can be converted to Address, chances are that it is
+        // TODO: What about H160 that are NOT addresses?
+        match Address::from_str(value.as_str().unwrap()) {
+            Ok(addr) => Some(ParamType::Address),
+            _ => Some(ParamType::String),
+        }
```

Comment:
here we probably need an untagged enum and try to deserialize it in the correct hex type like this

https://github.com/foundry-rs/foundry/blob/1bfb63906c2a96112cc9292d43c95e4c4a6169ab/common/src/fmt.rs#L252-L260